src/main/kotlin/pl/edu/agh/gem/external/persistence/GroupEntity.kt

```diff
@@ -0,0 +1,71 @@
+package pl.edu.agh.gem.external.persistence
+
+import org.springframework.data.annotation.Id
+import org.springframework.data.mongodb.core.index.Indexed
+import org.springframework.data.mongodb.core.mapping.Document
+import pl.edu.agh.gem.internal.model.Currency
+import pl.edu.agh.gem.internal.model.Group
+import pl.edu.agh.gem.internal.model.Member
+
+@Document("groups")
+data class GroupEntity(
+    @Id
+    val id: String,
+    val name: String,
+    val color: Long,
+    val ownerId: String,
+    val members: List<MemberEntity>,
+    val acceptRequired: Boolean,
+    val baseCurrency: List<CurrencyEntity>,
+    @Indexed(background = true, unique = true)
+    val joinCode: String,
+    val attachmentId: String,
+)
+
+data class MemberEntity(
+    val userId: String,
+)
+
+data class CurrencyEntity(
+    val code: String,
+)
+
+fun Group.toEntity() = GroupEntity(
+    id = id,
+    name = name,
+    color = color,
+    ownerId = ownerId,
+    members = members.map { it.toEntity() },
+    acceptRequired = acceptRequired,
+    baseCurrency = baseCurrency.map { it.toEntity() },
+    joinCode = joinCode,
+    attachmentId = attachmentId,
+)
+
+private fun Member.toEntity() = MemberEntity(
+    userId = userId,
+)
+
+private fun Currency.toEntity() = CurrencyEntity(
+    code = code,
+)
+
+fun GroupEntity.toDomain() = Group(
```

Comment:
shouldn't this function be a regular function (not extension) if we declare it GroupEntity.kt?

Reply:
It's an opinionated style-oriented rule to not have functions inside data classes

Reply:
But there's nothing inherently wrong with having methods in your data classes